fix(fib-heap): fix bug in merge fibonacci heap resolve #26

jonasjungaker opened the pull request:
Added check for when the merged heap is empty, the function does
nothing. When the original heap is empty, the funcitno simply
replaces the original heap with the merged heap and switches the
min_nodes. When both heaps are non-empty, the function links the
root lists and updates the min_node.

jonasjungaker committed:
fix(fib-heap): fix bug in merge fibonacci heap resolve #26
Added check for when the merged heap is empty, the function does
nothing. When the original heap is empty, the funcitno simply
replaces the original heap with the merged heap and switches the
min_nodes. When both heaps are non-empty, the function links the
root lists and updates the min_node.

jvonhacht commented on algorithms/heap/fibonacci_heap.py:
```diff
@@ -112,9 +112,15 @@ def merge(self, heap2):
+            self.min_node = heap2.min_node
+            return
+        tail = self.root_list.left
+        
```
Flytta heap2_tail under if satserna. Onödigt att allokera en variabel om den inte används i det fall vi går in i andra if-satsen.

jonasjungaker committed:
fix(fib-heap): refactor fib-heap merge #28
swapped places of the heap2_tail allocation in order to not
allocate a variable if it is not needed in the function execution
